pkg/auth/oauth2authenticator.go

```diff
@@ -359,50 +403,40 @@ func (o *oAuth2Authenticator) authenticateWithAuthorizationCode() error {
 		// launch browser
 		go o.openBrowserFunc(authCodeUrl)
 
+		// Handle context cancellation
 		timedOut := false
-		timer := time.AfterFunc(TIMEOUT_SECONDS, func() {
-			timedOut = true
+		canceled := false
+		go func() {
+			<-ctx.Done()
+			switch err := ctx.Err(); {
+			case errors.Is(err, context.DeadlineExceeded):
+				timedOut = true
+			case errors.Is(err, context.Canceled):
+				canceled = true
+			}
 			o.shutdownServerFunc(srv)
-		})
+		}()
+
 		listenErr = srv.Serve(listener)
 		if errors.Is(listenErr, http.ErrServerClosed) { // if the server was shutdown normally, there is no need to iterate further
+			if canceled {
+				o.logger.Info().Msg("OAuth2 request canceled.")
```

Comment:
could also be a timeout - the context could have been canceled or the timeout triggered.